### PR TITLE
Add ChangeLog entry for DTLS Connection ID

### PR DESCRIPTION
**Pull request by mpg**
We managed to forget it in https://github.com/Mbed-TLS/mbedtls/pull/6264

### CODE REVIEW

File: ChangeLog.d/dtls-connection-id.txt
Comment (AndrzejKurek) [inline ChangeLog.d/dtls-connection-id.txt:8]:
```suggestion
     disabled by default, and implement version 05 of the draft. It is now no
```
past

**Review by yanesca — DISMISSED** [review]
LGTM

**Review by AndrzejKurek — CHANGES_REQUESTED** [review]
Just one minor change requested.

**Review by AndrzejKurek — APPROVED** [review]

**Review by AndrzejKurek — APPROVED** [review]

**Review by yanesca — APPROVED** [review]
LGTM.

**Comment by mpg** [comment]
This PR only touches `ChangeLog.d`, and it passes Travis which includes `tests/scripts/all.sh -k 'check_*'` which are the only parts that care about files in this directory (`check_files` and `check_changelog` specifically), so we don't need to wait for the rest of the CI to complete before merging.

**Pull request merged**
